day 58 progress (#341)
saiyam1814/DevOpsRoadmap2022: atir-nayab/08-06-2022/scheduling.md

# what is kubernetes scheduling?

- the kubernetes scheduler is a core component of kubernetes: after a user or a controller creates a pod, the kubernetes scheduler, monitoring the object store for unassigned pods, will assign the pod to a node. Then, the kubelet, monitoring the object store for assingned pods, will execute the pod.

# what is the scheduler for?

![scheduler](../asset/scheduler.png)

the kubernetes scheduler is in charge of scheduling pods onto nodes. basically it works like this:

1. you create a pod

> 1. he scheduler notices that the new pod you created doesn't have a node assigned to it.

Reviewer: ```suggestion
1. The scheduler notices that the new pod you created doesn't have a node assigned to it.
```

> 1. the scheduler assigns a node tothe pod

Reviewer: ```suggestion
1. the scheduler assigns a node to the pod
```

# anti-node affinity?

- some scenarios require that you don't use one or more nodes except for particular pods. think of the nodes that host your monitoring application
- those nodes shouldn't have many resources due to the nature of their role. thus, if ohter pods than those which have the monitoring app are scheduled to those nodes, they hurt monitoring and also degrades the application they are hosting.
- in such a case, you need to use anti-affinity to keep pods away form a set of nodes.

# what is node taints and tolerations?

- this kubernetes feature allows users to mark a node (taint the node) so that no pods can be scheduled to it, unless a pod explicitly tolerates the taint.
- when you taint a node, it is automatically excluded from pod scheduling. when the schedule runs the predicate tests on a tainted node, thay'll fail unless the pod has toleration for that node.
- let assume new member joins the development team, writes a deployment for her application, but forgets to exclude the monitoring node from the target nodes? kubernetes administrators need a way to repel pods from nodes without having to modify every pod definition.

# tolerations

> - toleration is a way of ignoring a taint during scheduling. Tolerations aren't applied to nodes, but rather the pods. if we apply a toleration to teh podSpec, we could 'tolarate' the slow disks on that node and still use it.

Reviewer: ```suggestion
- toleration is a way of ignoring a taint during scheduling. Tolerations aren't applied to nodes, but rather the pods. if we apply a toleration to the podSpec, we could 'tolerate' the slow disks on that node and still use it.
```
